content/en/docs/concepts/workloads/controllers/jobs-run-to-completion.md

```diff
@@ -213,6 +213,10 @@ status check.
 {{< note >}}
 Issue [#54870](https://github.com/kubernetes/kubernetes/issues/54870) still exists for versions of Kubernetes prior to version 1.12
 {{< /note >}}
+{{< note >}}
+If your job have some error but your job restartPolicy is not `Never`, so your job created pods will be deleted, this logic make us find error log is difficult;
```

Comment:
A couple things... 

1) the created pods are not deleted even if restartPolicy is `OnFailure`. The container running the job will be restarted but the Pod won't be deleted.

2) I believe a more appropriate place for this note would be above after line 186.

I recommend rewording the note to the following:

```
If your job has `restartPolicy = "OnFailure"`, keep in mind that your container running the Job
will restart. This can make debugging the Job's executable more difficult. We suggest setting
`restartPolicy = "Never"` when debugging the Job or using a logging system to ensure output
from failed Jobs is not lost inadvertently.
```

Reply:
@jaypipes, any running job pods are deleted once the job's backoff limit has been reached. So if you have a restart policy of `OnFailure` once your pod has restarted 6 times (default backoff) it will be deleted so you can't inspect that pod. If your restart policy was `Never` you would instead have 6 pods all in a `Failed` pod phase that you could inspect. 

I think this is generally pretty unclear in the documentation, I've seen confusion around this before, might need a bigger change around clarifying when your job pods are terminated.

Reply:
OK, yeah, that's not entirely clear :) Thanks @Joseph-Irving. Do you have a suggestion of a better wording for us?

Reply:
my thinking is:
active pods will be deleted when job failed on job, but not related note on document.
so i want to add a note (some case job created pods will be deleted) remind other user.

i think `might need a bigger change around clarifying when your job pods are terminated` & `active pods will be deleted when job failed` is two things

@Joseph-Irving @jaypipes

Reply:
I think if you just make a slight change to @jaypipes suggestion it's more clear
```
If your job has `restartPolicy = "OnFailure"`, keep in mind that your container running the Job
will be terminated once the job backoff limit has been reached. This can make debugging the Job's executable more difficult. We suggest setting
`restartPolicy = "Never"` when debugging the Job or using a logging system to ensure output
from failed Jobs is not lost inadvertently.`
```
We(I?) could do a separate PR making it more implicit in the text when pods belonging to a job are terminated, I think the `Job Termination and Cleanup` just needs some reworking.

Reply:
That wording looks good to me. @fudali113 can you use that?

Reply:
ok, sorry for reply you on late
i rewrited it

@jaypipes